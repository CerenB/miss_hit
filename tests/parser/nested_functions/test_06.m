% From https://www.mathworks.com/help/matlab/matlab_prog/nested-functions.html

function test_06
    x = 5;
    z = nestfun;

    function y = nestfun
        y = x + 1;
    end
end
