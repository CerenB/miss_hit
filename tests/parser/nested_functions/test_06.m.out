----------------------------------------------------------------------
--  Parse tree for test_06.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test_06>
    Body: Sequence_Of_Statements
      Statements: Simple_Assignment_Statement
        LHS: Identifier <x>
        RHS: Number_Literal <5>
      Statements: Simple_Assignment_Statement
        LHS: Identifier <z>
        RHS: Identifier <nestfun>
    Nested: Function_Definition
      Signature: Function_Signature
        Name: Identifier <nestfun>
        Outputs: Identifier <y>
      Body: Sequence_Of_Statements
        Statements: Simple_Assignment_Statement
          LHS: Identifier <y>
          RHS: Binary_Operation +
            LHS: Identifier <x>
            RHS: Number_Literal <1>
----------------------------------------------------------------------
In test_06.m, line 7
|     function y = nestfun
|                ^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
